pyam/plotting.py

```diff
@@ -496,6 +497,67 @@ def bar_plot(df, x='year', y='value', bars='variable',
     return ax
 
 
+def boxplot(df, y='value', x='year', hue=None, palette='Spectral_r',
+            ax=None, legend=False, **kwargs):
+    """ Plot boxplot of data using seaborn.boxplot
+
+    Parameters
+    ----------
+    df : pandas.DataFrame or pyam.IamDataFrame
+        Data to plot as a long-form data frame
+    y : string, optional
+        The column to use for y-axis values representing the distribution
+        within the boxplot
+        default: 'value'
+    x : string, optional
+        The column to use for x-axis points, i.e. the number of boxes the plot
+        will have
+        default: 'year'
+    hue: string, optional
+        The column for grouping y-axis values at each x-axis point, i.e. a 3rd
+        dimension.
+        Data should be categorical, not a contiuous variable
+        default: None
+    palette:
+        Colour palette to use to differentiate between different caetgories in
+        hue
+        default: 'Spectral_r'
+    ax : matplotlib.Axes, optional
+    legend : bool, optional
+        Include a legend
+        default: False
+    kwargs : Additional arguments to pass to the pd.DataFrame.plot()
+    """
+
+    # Check data_type
+    if type(df) == pyam.core.IamDataFrame:
```

Comment:
all data preparation should go in the `box_plot()` function in `core.py`. see, e.g., https://github.com/IAMconsortium/pyam/blob/90bcef21019217d7030a5d94d8d142ad3fbdc6af/pyam/core.py#L1497